x/cash/staticfee_test.go

```diff
@@ -60,87 +44,92 @@ func must(obj orm.Object, err error) orm.Object {
 }
 
 func TestFees(t *testing.T) {
-	var helpers x.TestHelpers
 
 	cash := coin.NewCoin(50, 0, "FOO")
 	min := coin.NewCoin(0, 1234, "FOO")
 	perm := weave.NewCondition("sigs", "ed25519", []byte{1, 2, 3})
 	perm2 := weave.NewCondition("sigs", "ed25519", []byte{3, 4, 5})
 	perm3 := weave.NewCondition("custom", "type", []byte{0xAB})
 
-	cases := []struct {
+	cases := map[string]struct {
 		signers   []weave.Condition
 		initState []orm.Object
 		fee       *FeeInfo
 		min       coin.Coin
 		expect    checkErr
 	}{
-		// no fee given, nothing expected
-		0: {nil, nil, nil, coin.Coin{}, noErr},
-		// no fee given, something expected
-		1: {nil, nil, nil, min, errors.ErrInsufficientAmount.Is},
-		// no signer given
-		2: {nil, nil, &FeeInfo{Fees: &min}, min, errors.ErrInvalidInput.Is},
-		// use default signer, but not enough money
-		3: {
-			[]weave.Condition{perm},
-			nil,
-			&FeeInfo{Fees: &min},
-			min,
-			errors.ErrEmpty.Is,
+		"no fee given, nothing expected": {
```

Comment:
thanks for the cleanup.
one i found your map-driven (not table driven) tests with labels, there is no going back. love the descriptive names in the test errors (not test-4)

Reply:
I have learned it from one of the meetups in Berlin https://www.youtube.com/watch?v=yszygk1cpEc